jsonrpc/txpool_endpoint_test.go

```diff
@@ -1,58 +1,249 @@
+
+		assert.True(t, mockStore.includeQueued)
+		assert.Equal(t, 2, len(response.Pending))
+		assert.Equal(t, 2, len(response.Pending[address1]))
+		assert.Equal(t, 1, len(response.Pending[address2]))
+		assert.Equal(t, 2, len(response.Queued))
+	})
 }
 
 func TestInspectEndpoint(t *testing.T) {
-	dispatcher := newDispatcher(hclog.NewNullLogger(), newMockStore(), 0)
+	t.Run("returns empty InspectResponse if tx pool has no transactions", func(t *testing.T) {
+		mockStore := newMockTxPoolStore()
+		mockStore.maxSlots = 1024
+		txPoolEndpoint := &TxPool{mockStore}
+
+		result, _ := txPoolEndpoint.Inspect()
+		// nolint:forcetypeassert
+		response := result.(InspectResponse)
+
+		assert.True(t, mockStore.includeQueued)
+		assert.Equal(t, 0, len(response.Pending))
+		assert.Equal(t, 0, len(response.Queued))
+		assert.Equal(t, uint64(0), response.CurrentCapacity)
+		assert.Equal(t, mockStore.maxSlots, response.MaxCapacity)
+	})
+
+	t.Run("returns correct data for queued transactions", func(t *testing.T) {
+		mockStore := newMockTxPoolStore()
+		mockStore.capacity = 1
+		address1 := types.Address{0x1}
+		testTx := newTestTransaction(2, address1)
+		mockStore.queued[address1] = []*types.Transaction{testTx}
+		txPoolEndpoint := &TxPool{mockStore}
+
+		result, _ := txPoolEndpoint.Inspect()
+		// nolint:forcetypeassert
+		response := result.(InspectResponse)
+
+		assert.Equal(t, 0, len(response.Pending))
+		assert.Equal(t, 1, len(response.Queued))
+		assert.Equal(t, uint64(1), response.CurrentCapacity)
+		transactionInfo := response.Queued[testTx.From.String()]
+		assert.NotNil(t, transactionInfo)
+		assert.NotNil(t, transactionInfo[strconv.FormatUint(testTx.Nonce, 10)])
+	})
 
-	resp, err := dispatcher.Handle([]byte(`{
-		"method": "txpool_inspect",
-		"params": []
-	}`))
-	assert.NoError(t, err)
+	t.Run("returns correct data for pending transactions", func(t *testing.T) {
+		mockStore := newMockTxPoolStore()
+		mockStore.capacity = 2
+		address1 := types.Address{0x1}
+		testTx := newTestTransaction(2, address1)
+		testTx2 := newTestTransaction(3, address1)
+		mockStore.pending[address1] = []*types.Transaction{testTx, testTx2}
+		txPoolEndpoint := &TxPool{mockStore}
 
-	var res InspectResponse
+		result, _ := txPoolEndpoint.Inspect()
+		// nolint:forcetypeassert
+		response := result.(InspectResponse)
 
-	assert.NoError(t, expectJSONResult(resp, &res))
-	assert.Len(t, res.Pending, 0)
-	assert.Len(t, res.Queued, 0)
-	assert.Equal(t, res.CurrentCapacity, uint64(0))
-	assert.Equal(t, res.MaxCapacity, uint64(0))
+		assert.Equal(t, 1, len(response.Pending))
+		assert.Equal(t, 0, len(response.Queued))
+		assert.Equal(t, uint64(2), response.CurrentCapacity)
+		transactionInfo := response.Pending[testTx.From.String()]
+		assert.NotNil(t, transactionInfo)
+		assert.NotNil(t, transactionInfo[strconv.FormatUint(testTx.Nonce, 10)])
+		assert.NotNil(t, transactionInfo[strconv.FormatUint(testTx2.Nonce, 10)])
+	})
 }
 
 func TestStatusEndpoint(t *testing.T) {
```

Comment:
Small nitpick, I really like the convection that we have introduced in eth_blockchain_test.go. To name every test TestEth_, I would rename these methods TestTxPool_Status. The only benefit of this change would be an easier search.